WordPressShared/WordPressShared/NSDate+Helpers.swift

```diff
@@ -13,6 +13,14 @@ extension Date {
             return formatter
         }()
 
+        static let iso8601WithMilliseconds: DateFormatter = {
+            let formatter           = DateFormatter()
+            formatter.locale        = Locale(identifier: "en_US_POSIX")
+            formatter.dateFormat    = "yyyy-MM-dd'T'HH:mm:ss.SSSXXXXX"
```

Comment:
please let me know if there would be a more helpful format for us to send from the server here. Javascript in this case I think will be much more flexible than iOS.

Reply:
This format is OK as long as we stick with the standard.

Reply:
wow. I was hoping that something like [NSISO8601DateFormatter](https://developer.apple.com/documentation/foundation/nsiso8601dateformatter) would help here but it looks like it's riddled with issues on fractional seconds. Namely, maybe it wants to match in the format string the precision being supplied in the string?

Makes no sense to me. There's a [fractional seconds option](https://developer.apple.com/documentation/foundation/nsiso8601dateformatoptions/nsiso8601dateformatwithfractionalseconds?language=objc) but no examples so I don't know what to make of it.

Could you do me a favor and quell my curiosity? If you supply the same time string but put six digits after the decimal for seconds (microseconds) with `ss.SSS` can you tell me if it _fails_ or if it truncates/rounds to three digits of precision?

Reply:
I didn't use the fractional seconds option because it's only available starting on iOS 11 and we are still supporting iOS 10.

> if you supply the same time string but put six digits after the decimal for seconds (microseconds) with ss.SSS can you tell me if it fails or if it truncates/rounds to three digits of precision?

It works, the documentation claims it truncates, and a quick test indicates that's correct.